scripts/pfcoreUtils.js

```diff
@@ -0,0 +1,188 @@
+const outputFolderSetup = async () => {
+  try {
+    if (!fs.existsSync(outputFolder)) {
+      fs.mkdirSync(outputFolder)
+    }
+    if (!fs.existsSync(`${outputFolder}/jarsAndFarms.ts`)) {
+      fs.closeSync(fs.openSync(`${outputFolder}/jarsAndFarms.ts`, 'a'))
+    }
+    if (!fs.existsSync(`${outputFolder}/implementations`)) {
+      fs.mkdirSync(`${outputFolder}/implementations`)
+    }
+    if (!fs.existsSync(`${outputFolder}/implementations/${pfcoreArgs.chain}`)) {
+      fs.mkdirSync(`${outputFolder}/implementations/${pfcoreArgs.chain}`)
+    }
+    if (!fs.existsSync(`${outputFolder}/jarBehaviorDiscovery`)) {
+      fs.mkdirSync(`${outputFolder}/jarBehaviorDiscovery`)
+    }
+    if (!fs.existsSync(`${outputFolder}/jarBehaviorDiscovery/modelImports.ts`)) {
+      fs.closeSync(fs.openSync(`${outputFolder}/jarBehaviorDiscovery/modelImports.ts`, 'a'))
+    }
+    if (!fs.existsSync(`${outputFolder}/jarBehaviorDiscovery/implImports.ts`)) {
+      fs.closeSync(fs.openSync(`${outputFolder}/jarBehaviorDiscovery/implImports.ts`, 'a'))
+    }
+    if (!fs.existsSync(`${outputFolder}/jarBehaviorDiscovery/jarToBehaviorSet.ts`)) {
+      fs.closeSync(fs.openSync(`${outputFolder}/jarBehaviorDiscovery/jarToBehaviorSet.ts`, 'a'))
+    }
+  } catch (err) {
+    console.error(err)
+  }
+}
+
+const incrementJar = async (jarCode, index) => {
+  const nextLetter = letter => {
+    let charCode = letter.charCodeAt(0);
+    return String.fromCharCode((charCode - 96) % 26 + 97)
+  };
+  const lastLetter = jarCode.split('').pop();
+  if (lastLetter === 'z') {
+    pfcoreArgs.jarCode = jarCode.split('').slice(0, jarCode.length - 1).concat(nextLetter(lastLetter)).join('');
+    pfcoreArgs.jarCode += "a";
+  } else if (index != 0) {
+    pfcoreArgs.jarCode = jarCode.split('').slice(0, jarCode.length - 1).concat(nextLetter(lastLetter)).join('')
+  }
+};
+
+const generateJarBehaviorDiscovery = async (args) => {
```

Comment:
The code here is a little hard to read but it's not bad because any code generating other code is always a bit tough. That's just the way it is. So this is mostly ok.